gobblin-aws/src/main/java/gobblin/aws/GobblinAWSClusterLauncher.java

```diff
@@ -0,0 +1,669 @@
+        .withValue(this.clusterName);
+    final List<AutoScalingGroup> autoScalingGroups = this.awsSdkClient.getAutoScalingGroupsWithTag(clusterNameTag);
+
+    // If no auto scaling group is found, we don't have an existing cluster to connect to
+    if (autoScalingGroups.size() == 0) {
+      return Optional.absent();
+    }
+
+    // If more than 0 auto scaling groups are found, validate the setup
+    if (autoScalingGroups.size() != 2) {
+      throw new IOException("Expected 2 auto scaling groups (1 each for master and workers) but found: " +
+        autoScalingGroups.size());
+    }
+
+    // Retrieve cluster information from ASGs
+    Optional<String> clusterId = Optional.absent();
+    Optional<AutoScalingGroup> masterAsg = Optional.absent();
+    Optional<AutoScalingGroup> workersAsg = Optional.absent();
+
+    for (TagDescription tagDescription : autoScalingGroups.get(0).getTags()) {
+      LOGGER.info("Found tag: " + tagDescription);
+      if (tagDescription.getKey().equalsIgnoreCase(CLUSTER_ID_ASG_TAG)) {
+        clusterId = Optional.of(tagDescription.getValue());
+      }
+      if (tagDescription.getKey().equalsIgnoreCase(ASG_TYPE_ASG_TAG)) {
+        if (tagDescription.getValue().equalsIgnoreCase(ASG_TYPE_MASTER)) {
+          masterAsg = Optional.of(autoScalingGroups.get(0));
+          workersAsg = Optional.of(autoScalingGroups.get(1));
+        } else {
+          masterAsg = Optional.of(autoScalingGroups.get(1));
+          workersAsg = Optional.of(autoScalingGroups.get(0));
+        }
+      }
+    }
+
+    if (!clusterId.isPresent()) {
+      throw new IOException("Found 2 auto scaling group names for: " + this.clusterName +
+          " but tags seem to be corrupted, hence could not determine cluster id");
+    }
+
+    if (!masterAsg.isPresent() || !workersAsg.isPresent()) {
+      throw new IOException("Found 2 auto scaling group names for: " + this.clusterName +
+      " but tags seem to be corrupted, hence could not determine master and workers ASG");
+    }
+
+    // Get Master and Workers launch config name and auto scaling group name
+    this.masterAutoScalingGroupName = masterAsg.get().getAutoScalingGroupName();
+    this.masterLaunchConfigName = masterAsg.get().getLaunchConfigurationName();
+    this.workerAutoScalingGroupName = workersAsg.get().getAutoScalingGroupName();
+    this.workerLaunchConfigName = workersAsg.get().getLaunchConfigurationName();
+
+    LOGGER.info("Trying to find cluster master public ip");
+    this.masterPublicIp = getMasterPublicIp();
+    LOGGER.info("Master public ip: "+ this.masterPublicIp);
+
+    return clusterId;
+  }
+
+  /**
+   * Setup the Gobblin AWS cluster.
+   *
+   * @throws IOException If there's anything wrong setting up the AWS cluster
+   */
+  @VisibleForTesting
+  String setupGobblinCluster() throws IOException {
+
+    final String uuid = UUID.randomUUID().toString();
+
+    // Create security group
+    // TODO: Make security group restrictive
```

Comment:
Nothing. Today we open more than required ports, however, we need to restrict it to only that are needed by Zookeeper / Helix, UI, and Kibana and other modules going forward